gulpfile.js

```diff
@@ -93,6 +98,39 @@ gulp.task('compile_test', function() {
     .pipe(buildTest())
 });
 
+/**
+ * Task only used to verify TypeScript autogenerated typings.
+ */
+gulp.task('compile_autogen_typing', function() {
+  return gulp.src(paths.src)
+    // Compile Typescript into .js and .d.ts files
+    .pipe(buildAutogenTyping())
+
+    // Exclude typings that are unintended (only RTDB and FirebaseApp are
+    // expected to produce typings).
+    .pipe(filter([
+      'lib/src/**/*.js',
+      'lib/src/**/*.d.ts',
+      '!lib/src/index.d.ts',
+      '!lib/src/default-namespace.d.ts',
+      '!lib/src/firebase-namespace.d.ts',
+      '!lib/src/firebase-service.d.ts',
+      '!lib/**/*-internal.d.ts',
```

Comment:
Let's add some comments here to provide more context. We remove `-internal.d.ts` stuff because we don't ever want them exposed. We delete the other stuff *for now* since they are a work in progress.